actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -7,7 +7,12 @@ def send_action(method, *args)
     end
 
     def default_render(*args)
-      render(*args)
+      if template_exists?(action_name.to_s, _prefixes, variants: request.variant)
```

Comment:
Now we're checking if the template exists twice, if the controller doesn't define the action and we hit `method_for_action`. The exists call is probably cached and I guess that's an exceptional state.